core/src/sumeragi/mod.rs

```diff
@@ -205,33 +205,32 @@ impl SumeragiHandle {
             kura,
             network,
             genesis_network,
-            block_hashes,
+            block_count,
         }: SumeragiStartArgs,
     ) -> SumeragiHandle {
         let (message_sender, message_receiver) = mpsc::sync_channel(100);
 
-        for (block_hash, i) in block_hashes
-            .iter()
-            .take(block_hashes.len().saturating_sub(1))
-            .zip(1u64..)
-        {
-            let block_height: u64 = i;
-            let block_ref = kura.get_block_by_height(block_height).expect("Sumeragi could not load block that was reported as present. Please check that the block storage was not disconnected.");
-            assert_eq!(
-                block_ref.hash(),
-                *block_hash,
-                "Kura init correctly reported the block hash."
-            );
-
-            wsv.apply(&block_ref)
+        let mut blocks_iter = (1..=block_count).map(|block_height| {
+            kura.get_block_by_height(block_height as u64).expect("Sumeragi could not load block that was reported as present. Please check that the block storage was not disconnected.")
```

Comment:
`.expect()` error messages should express not the error that happened, but the condition or assumption that was **expected** to be valid.

I would rewrite it to

> Sumeragi should be able to load the block that was reported as presented. If not, the block storage was probably disconnected.

Not sure about using hints like "please check" is appropriate for panic messages. If advanced error handling is needed, it should be a `Result`, not `panic!` (not sure).

Reply:
I agree, implemented your changes.